src/client/boundaries.rs

```diff
@@ -677,27 +838,29 @@ impl Boundaries {
             boundary_characters: HashMap::new(),
         }
     }
-    pub fn add_rect(&mut self, rect: &dyn Pane) {
+    pub fn add_rect(&mut self, rect: &dyn Pane, should_be_colored: bool, input_mode: InputMode) {
         if rect.x() > 0 {
             let boundary_x_coords = rect.x() - 1;
             let first_row_coordinates = self.rect_right_boundary_row_start(rect);
             let last_row_coordinates = self.rect_right_boundary_row_end(rect);
             for row in first_row_coordinates..last_row_coordinates {
                 let coordinates = Coordinates::new(boundary_x_coords, row);
                 let mut symbol_to_add = if row == first_row_coordinates && row != 0 {
-                    BoundarySymbol::new(boundary_type::TOP_LEFT)
+                    BoundarySymbol::new(boundary_type::TOP_LEFT, should_be_colored)
                 } else if row == last_row_coordinates - 1 && row != self.rows - 1 {
-                    BoundarySymbol::new(boundary_type::BOTTOM_LEFT)
+                    BoundarySymbol::new(boundary_type::BOTTOM_LEFT, should_be_colored)
                 } else {
-                    BoundarySymbol::new(boundary_type::VERTICAL)
+                    BoundarySymbol::new(boundary_type::VERTICAL, should_be_colored)
```

Comment:
Also, maybe we can make the method assigning color to `BoundarySymbol` a Builder type method? Something like:
```
BoundarySymbol::new(boundary_type::VERTICAL).color(colors::GREEN)
```
What do you think? I just think it will be more idiomatic rust.

Reply:
That's definitely something I can do